libexec/basher-update

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+#
+# Summary: Checks all packages for updates
+# Usage: basher update
+
+set -e
+
+shopt -s nullglob
+
+IFS=$'\r\n' packages=($(basher list))
```

Comment:
Should `IFS=$'\r\n'` be on its own line to make clear it changes `IFS` for the rest of the script, or should a `temp_ifs`/`old_ifs`-like variable be used just for `packages=()`?

Reply:
I'm gonna be honest, I took that part as a pure copy/paste from basher-outdated and I just changed the inner loop:

```bash
printf "update $package ...            \r"
basher-_unlink "$package"
git pull &> /dev/null
basher-_link "$package"
```

I added the `printf` statements so you can follow what it is doing. 
Otherwise it was just 20 seconds of waiting without a clue what was happening.

Reply:
That's a good idea. If that `IFS` thing is in other places, it might be worth fixing that separately